File: pset1/mario/mario.c
#include <cs50.h>
#include <stdio.h>

void print(char c, int times); // beautify the program by declaring the function

int main()
{
    int num;
    do
    {
        num = get_int("Height: "); // input the height
    }
    while (num < 1 || num > 8);

    for (int i = 0; i < num; i++)
    {
        print(' ', num - i - 2);
        print('#', i);
        printf("  ");
        print('#', i);
        printf("\n");
    }
}

void print(char c, int times)    // function to print the charachters in a loop
{
    while (times >= 0)
    {
        printf("%c", c);
        times--;
    }
}

File: pset6/mario/more/mario.py
from cs50 import get_int


# pyramid function
def pyramid(num):
    for i in range(1, num + 1):
        print(" "*(num - i), "#"*i, "  ", "#"*i, sep="")


# input loop
while True:
    num = get_int("Height: ")
    if num in range(1, 9):
        break

# function calling
pyramid(num)
